pokete_classes/landscape.py

```diff
@@ -26,18 +27,28 @@ def action(self, ob):
                  or (not n_a and not is_night)
                  or (n_a and is_night)}
         if random.randint(0, 8) == 0:
-            fm.fight(Poke("__fallback__", 0)
-                     if len([poke for poke in self.figure.pokes[:6]
-                             if poke.hp > 0]) == 0
-                     else
-                     [poke for poke in self.figure.pokes[:6] if poke.hp > 0][0],
-                     Poke(random.choices(list(pokes),
-                                         weights=[i["rarity"] for _, i in
-                                                  pokes.items()])[0],
-                          random.choices(list(range(self.arg_proto["minlvl"],
-                                                    self.arg_proto["maxlvl"])))[
-                              0],
-                          player=False, shiny=(random.randint(0, 500) == 0)))
+            fm.fightmap.fight(
+                [
+                    self.figure,
+                    NatureProvider(
+                        Poke(
+                            random.choices(
+                                list(pokes),
+                                weights=[i["rarity"] for _, i in pokes.items()]
```

Comment:
No point looping over `.items()` and then ignoring the key, you're wasting tuple constructions
```py
weights=[i["rarity"] for i in pokes.values()]
```